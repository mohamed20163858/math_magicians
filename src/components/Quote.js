const Quote = () => (
  <p data-testid="momo-id">
    {' '}
    Mathematics is not about numbers, equations, computations, or algorithms:
    it is about understanding. - William Paul Thurston
  </p>
);

export default Quote;
