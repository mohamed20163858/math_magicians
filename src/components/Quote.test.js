import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Quote from './Quote';

it('renders correctly', () => {
  const tree = renderer
    .create(<Quote />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('existing of quote in page ', () => {
  // ARRANGE
  render(<Quote />);

  // ACT
  const result = screen.getByTestId('momo-id');
  // ASSERT
  expect(result).toBeInTheDocument();
});
