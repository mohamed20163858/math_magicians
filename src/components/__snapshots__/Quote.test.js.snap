// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<p
  data-testid="momo-id"
>
   
  Mathematics is not about numbers, equations, computations, or algorithms: it is about understanding. - William Paul Thurston
</p>
`;
